openpgp/packet/signature.go

```diff
@@ -66,7 +66,7 @@ type Signature struct {
 
 	SigLifetimeSecs, KeyLifetimeSecs                        *uint32
 	PreferredSymmetric, PreferredHash, PreferredCompression []uint8
-	PreferredAEAD                                           []uint8
+	PreferredCipherSuite                                    []uint8
```

Comment:
I would make this 
```suggestion
	PreferredCipherSuites                                   [][2]uint8
```
(The ones above are singular but have an implicit "Algorithms" after them, this one doesn't).

And then, initializing it from a 1D array is a bit more work, but then once it's a 2D array it should be easier to use.
And for going back from 2D -> 1D for serialization, you could use `bytes.Join(sig.PreferredCipherSuites, nil)`.

Reply:
Should I use `[] CipherSuite` maybe?

Reply:
Seems like you can't convert a 2d-array into a 2d-slice, therefore the solution 
```
bytes.Join(sig.PreferredCipherSuites, nil)
```
does not work.

So far I'd propose to use
```
serialized := make([]byte, len(sig.PreferredCipherSuites)*2)
for i, cipherSuite := range sig.PreferredCipherSuites {
	serialized[2*i] = cipherSuite[0]
	serialized[2*i+1] = cipherSuite[1]
}
```
Let me know if this looks good to you. In this case switching to `[] CipherSuite` does not add overhead.
Other option is to use slices directlu and declare it as `[][]uint8`

Reply:
I see, OK. Yeah, then I would go with serializing manually (and keeping it as `[][2]uint8`, because the other types also don't use `packet.CipherFunction` etc.)